.circleci/config.yml

```diff
@@ -4,32 +4,33 @@ jobs:
   backend:
     docker:
       - image: cimg/python:3.12
-    environment:
-      PIPENV_VENV_IN_PROJECT: true
     steps:
       - checkout
       - restore_cache:
-          key: pipenv-v3-{{ checksum "setup.py" }}
+          key: &venv-cache venv-v1-{{ checksum "setup.py" }}
       # Only install if .venv wasn’t cached.
       - run: |
           if [[ ! -e ".venv" ]]; then
-              pipenv install -e .[testing,docs]
+              python -m venv .venv
+              source .venv/bin/activate
+              python -m pip install -e .[testing,docs]
           fi
+          echo "source $(pwd)/.venv/bin/activate" >> $BASH_ENV
```

Comment:
Each step in a job spawns a new shell, so we need to activate the venv every time. Instead of adding it to each job, we add this to `$BASH_ENV` so it's done automatically

https://discuss.circleci.com/t/activate-python-virtualenv-for-whole-job/14434